### PR TITLE
Project Complete

### PR DESCRIPTION
**Pull request by s0kil**
- Added the User model
- `index`, `edit`, `create` controller actions for the User model
- `index`, `edit`, `create` views created

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    if @user.save
+      redirect_to new_users_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    console
```
Comment (trekab) [inline app/controllers/users_controller.rb]: Remove the `console` command. It is not relevant after development(in production code) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/debugging_requirement.png) from Pathwright)

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Hi @s0kil and @e71az! You have done a great job following the project instructions as laid out in the Odin project. There are a few details I'd like you to attend to before moving forward. These are:
- [x] Your `Getting Started` instructions in the `README.md` file should include `bundle install` and `rails db:migrate`commands since your app can't work out of the box without the necessary gems installed and migrates run.
- [x] On creating a new user, I get this error: `undefined local variable or method ``new_users_path' for #<UsersController:0x000055d64dc61258> Did you mean? new_user_path`. Please fix this.
- [x] __[Optional]__ Create a partial for the user form and reuse it in the `new` and `edit` views. This helps in reducing repeated code in your application.
- [x] __[Optional]__ Great job implementing error messages for the `edit` view. You could do the same for the `new` view. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
- [x] __[Optional]__ Validate the email using a regular expression. This is a good place to practice this kind of validation since it is the industry standard.
- [x] Remove the "console" command from the code as indicated in the comment in the code below.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Comment by eliasecasta** [comment]
> ### Changes Required! ♻️
> Hi @s0kil and @e71az! You have done a great job following the project instructions as laid out in the Odin project. There are a few details I'd like you to attend to before moving forward. These are:
> 
> * [x]  Your `Getting Started` instructions in the `README.md` file should include `bundle install` and `rails db:migrate`commands since your app can't work out of the box without the necessary gems installed and migrates run.
> * [x]  On creating a new user, I get this error: `` undefined local variable or method ``new_users_path' for #<UsersController:0x000055d64dc61258> Did you mean? new_user_path ``. Please fix this.
> * [x]  **[Optional]** Create a partial for the user form and reuse it in the `new` and `edit` views. This helps in reducing repeated code in your application.
> * [x]  **[Optional]** Great job implementing error messages for the `edit` view. You could do the same for the `new` view. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
> * [x]  **[Optional]** Validate the email using a regular expression. This is a good place to practice this kind of validation since it is the industry standard.
> * [x]  Remove the "console" command from the code as indicated in the comment in the code below.
> 
> **Kindly make the necessary changes and submit another code review request.**
> _Feel free to reach me on slack for any inquiries about this review._ 😉
> **Treasure Kabareebe**
> [GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

Thank you for your review @trekab, we applied all the changes that you requested.

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  <br>
+  <%= f.label :email, "E-mail:" %>
+  <%= f.email_field :email, placeholder:"example@example.com" %>
+  <br>
+  <%= f.label :password, "Password:" %>
+  <%= f.password_field :password, placeholder:"12345" %>
+  <br>
+  <br>
+  <%= f.submit %>
+<% end %>
```
Comment (addod19) [inline app/views/users/new.html.erb]: Kindly remove all commented code :+1:

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  <%= label_tag(:email, "E-mail:") %>
+  <%= email_field_tag(:email) %>
+  <br>
+  <%= label_tag(:password, "Password:") %>
+  <%= password_field_tag(:password) %>
+  <br>
+  <br>
+  <%= submit_tag("Create user") %>
+<% end %>
+-->
```
Comment (addod19) [inline app/views/users/new.html.erb]: Kindly remove commented code :+1:

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  <label for="user[email]">Email:</label>
+  <input type="email" name="user[email]">
+  <br>
+  <label for="user[password]">Password:</label>
+  <input type="password" name="user[password]">
+
+  <hr>
+  <input type="submit" value="Create User">
+</form>
+-->
```
Comment (addod19) [inline app/views/users/new.html.erb]: Kindly remove commented code :+1:

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi **TEAM**, great job on this project :fireworks: Having fixed all the comments outlined by the previous TSE, there is one issue that prevent us from approving the project. Kindly check the comments below: :smile: 
- Kindly remove all comments in our code

Once this is fixed, kindly submit another code review :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](linkedin.com/in/daniel-larbi-addo/)

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi **TEAM**, congratulations on making all the changes outlined by the previous TSE :+1: Our project now meets all the minimum requirements :smile: 

Go ahead and merge into development :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**